src/main/java/net/fabricmc/loader/api/metadata/ModMetadata.java

```diff
@@ -63,6 +64,7 @@
 
 	boolean containsCustomValue(String key);
 	CustomValue getCustomValue(String key);
+	Map<String, CustomValue> getCustomValues();
```

Comment:
Wouldn't it make sense to return a `CvObject` here ?

Reply:
I think it is possible to define any custom value there, such as:
```json
"custom": {
  "myval": true
}
```

Which wouldn't be a CvObject

Reply:
@i509VCB He means to return `CvObject getCustomValues()` than `Map<String, CustomValue> getCustomValues()`

Reply:
Oh I see what was meant. Yeah that makes sense since the root value must be an object

Reply:
I would have thought that the `Map` is likely to be more useful to expose than a `CvObject`. (Although I think this brings up the related question of why CvObject doesn't implement Map directly).

Reply:
Imo `Map` is bad, as we should not expose any modification operations. `CvObject` doesn't while `Map` does.

Reply:
My reasoning is that we have a vast amount of library methods that can operate on maps directly (as it's a standard Java collection), plus it has a few more methods that might be useful to users (such as `keySet()`), whereas `CvObject` is a minimal interface with no javadoc.

Reply:
I could see adding a keySet() method or making it Iterable as of great benefit in another PR. Shouldn't be too bad in the impl side since CvObjectImpl is backed by a map.

Reply:
With your reasoning, we should dump the whole custom value system in favor of plain Object, and use checks like `instanceof Number` `instanceof Boolean` `instanceof Map` to retrieve the sub objects.

I personally disagree with your proposal. The whole custom value system uses custom value, if they can use a map for the top level they still can't avoid depending on/using custom values in lower levels, so there is no point.

Reply:
No, that was not my suggestion. We still need `CustomValue` to limit the types that are available lower-down, however `Map<String, CustomValue>` *does* limit the possible types down to what can be expressed in JSON.

Reply:
Oh yeah, taking AlexIIL's suggestions into consideration, this doesn't solve much unless we make CvObject implement `Iterable<CustomValue>`

So I'll need to add methods to CvObject to expose the child custom objects

Reply:
Edit2: Wait no it is iterable, what was I thinking lol

Reply:
CvObject is already Iterable<Map.Entry<String, CustomValue>>

Reply:
The root type is always going to be a JSON object, which is exactly a `CvObject`